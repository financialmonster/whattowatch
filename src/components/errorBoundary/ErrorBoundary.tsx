import React, { ReactNode } from 'react';
import { Message } from 'components/message/Message';

type TErrorBoundaryProps = {
    children: ReactNode;
}

type TErrorBoundaryState = {
    isError: boolean;
}

export class ErrorBoundary extends React.PureComponent<TErrorBoundaryProps, TErrorBoundaryState> {
    state = {
        isError: false
    }

    static getDerivedStateFromError() {
        return { isError: true};
    }

    render() {
        if(this.state.isError) {
            return (
                <Message>
                    We are sorry. Our application isn't available now. Please, reload the page.
                </Message>
            );
        }

        return this.props.children;
    }
}
