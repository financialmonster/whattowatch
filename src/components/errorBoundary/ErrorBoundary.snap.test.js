import React from 'react';
import renderer from 'react-test-renderer';

import { ErrorBoundary } from './ErrorBoundary';

it(`ErrorBoundary: correctly rendered`, () => {
    const tree = renderer.create(
        <ErrorBoundary>
            Test
        </ErrorBoundary>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
