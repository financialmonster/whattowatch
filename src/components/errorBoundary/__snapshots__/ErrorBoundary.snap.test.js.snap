// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary: correctly rendered 1`] = `"Test"`;
